packages/node/src/indexer/indexer.manager.ts

```diff
@@ -171,17 +175,37 @@ export class IndexerManager {
 
   private async ensureMetadata(schema: string) {
     const metadataRepo = MetadataFactory(this.sequelize, schema);
-    //block offset should only been create once, never update.
-    //if change offset will require re-index and re-sync poi
-    const blockOffset = await metadataRepo.findOne({
-      where: { key: 'blockOffset' },
+
+    const networkMeta = this.apiService.getNetworkMetadata();
+
+    console.log('here');
+
+    const count = await metadataRepo.count({
+      where: {
+        key: [
+          'blockOffset',
+          'indexerNodeVersion',
+          'chain',
+          'specName',
+          'genesisHash',
+        ],
+      },
     });
-    if (!blockOffset) {
+
+    if (count < 5) {
+      //block offset should only been create once, never update.
+      //if change offset will require re-index and re-sync poi
       const offsetValue = (this.getStartBlockFromDataSources() - 1).toString();
-      await metadataRepo.create({
-        key: 'blockOffset',
-        value: offsetValue,
-      });
+
+      await Promise.all([
```

Comment:
I could have made a method that takes a list of metadata key/value pairs to be stored. Should I save this for the next pull request adding batch calls?.

Reply:
I'm not so worried about batch calls.
There's still something wrong about the logic here. For `blockOffset` and `genesisHash`, they are set when empty, but don't use `count` because you must know which one of them are empty and not updating the other one.
For `indexerNodeVersion`, `chain` & `specName` they can be overwritten to different value, and you can compare first and only update when value is different.

Reply:
I have updated this.